### PR TITLE
some feature update

### PR DESCRIPTION
**Pull request by yjf2002ghty**
demo:  [https://fairyground-new.vercel.app/](https://fairyground-new.vercel.app/)

### **Minor changes:**

1. added box for Threads, Hash, MultiPV
2. added info section to show current board FEN (useful for copying the game to other applications)
3. a pop up text to show the game reult

### **Major changes**

1. added support for promotion and gating(for wall squares, seirawan gating are considered as promotion.) by making users input their choice in a browser prompt dialog
2. added support for viewing previous positions in the game. This allows user to analyze previous situations

Minor updates are small improvements so they don't have bugs (At least I haven't experienced any bugs).
Major updates are large improvements that can create some hidden bugs or contain some not well-written codes. They need further maintaining.

### CODE REVIEW

**Comment by yjf2002ghty** [comment]
I've also removed "play" and "analysis" page as their functions are highly duplicated with "advanced" page.

**Comment by ianfab** [comment]
Thanks a lot! The support for promotions and gating is a really nice and very much needed addition.

To review it in detail will take more time. One comment though regarding the extra pages. What you now renamed to the index page was the only page originally and only recently I added the additional pages. My main goal was to simplify the interface, since fairyground is currently the only real link I have from the official website to quickly try out live what Fairy-SF is about, so it should in my mind more or less just take selecting a variant to immediately start playing or analyzing. However, I did not want to drop the original layout, because I still found it useful for expert users. So to me it would still be important to somehow keep both target groups in mind.

**Comment by yjf2002ghty** [comment]
supporttheme
supportwall

**Major update**
Now supports theme per variant & wall square display.

**Comment by ianfab** [comment]
Wow, thanks. I really like that the customization of the themes is configuration and not hard-coded, that makes it really flexible.

I would just have two small requests:
* Please run the formatting
* It looks like the SVG/CSS generation script is outdated now. Could you please update it or alternatively document how you generated the files?

**Comment by ianfab** [comment]
Thanks for all the additions. Are you planning to do further changes, or this PR ready for review?

**Comment by ianfab** [comment]
Just to check back, are you planning to do any further changes to the PR and/or address my comments from above, or should I take over from here?

File: src/main.js [hunk 1]
```diff
@@ -371,17 +414,49 @@ function afterChessgroundMove(orig, dest, metadata) {
...
     promotion = choice;
   }
 
-  if (!board.push(move + promotion)) {
+  choice = null;
+
+  if (possiblegatings.length > 1)  //if there are more than one option
+  {
+    while (true) {
+      choice = prompt(`There are multiple chioces that you can gate a wall square. They are\n${possiblegatings}\n, where = means do not gate, letters with numbers mean destination square (e.g. e4e5 means your piece has moved to e4 and you gate a wall square to e5). Now please enter your choice: `,"");
```
Comment (megaminxcuber) [inline src/main.js]: chioces should be choices

**Pull request merged**